Index: qemu-kvm-0.12.3/hw/arm_boot.c
===================================================================
--- qemu-kvm-0.12.3.orig/hw/arm_boot.c	2010-03-20 10:31:24.200423864 +0100
+++ qemu-kvm-0.12.3/hw/arm_boot.c	2010-03-20 10:31:38.120443588 +0100
@@ -15,7 +15,7 @@
 
 #define KERNEL_ARGS_ADDR 0x100
 #define KERNEL_LOAD_ADDR 0x00010000
-#define INITRD_LOAD_ADDR 0x00800000
+#define INITRD_LOAD_ADDR 0x00d00000
 
 /* The worlds second smallest bootloader.  Set r0-r2, then jump to kernel.  */
 static uint32_t bootloader[] = {
